### DISCUSSION TITLE
Wait for the task to complete

### DISCUSSION THREAD

**Discussion by venumanthena** [gilbarbara/react-joyride#648] [Q&A]
Thanks for the fantastic library. However,  I had a doubt.

I have a steps json in which few actions need to be forced to click. On clicking, the next slide should appear once the task data is loaded or in other words, action is completed. So what is happening here is that by the time my data is loaded, the steps are firing one by one. So the data is always returning false. I can workaround with this by increasing the stepindex in my data callback functions but I wanted a generic solution so that i can pass a key and based on that key, the action moves to next slide. Hope I am clear with my explanation. Below is my steps JSON

           steps={[{
                target: "#generatReportBtn",
                disableBeacon: true,
                spotlightClicks: true,
                strict: true,
                actionCompleted: !isEmpty(get(reportData, "data", null)),
                styles: {
                  options: {
                    zIndex: 100000,
                  },
                },
                floaterProps: {
                  disableAnimation: true,
                },
                content: {
                  title: "Generate Report",
                  desc: "Generate Report button",
                },
              },
              {
                target: "#saveReportBtn",
                spotlightClicks: false,
                strict: false,
                styles: {
                  options: {
                    zIndex: 100000,
                  },
                },
                disableBeacon: true,
                floaterProps: {
                  disableAnimation: true,
                },
                content: {
                  title: "Save Report",
                  desc: "Save Report button",
                },
              },
              {
                target: "#quickAccess",
                styles: {
                  options: {
                    zIndex: 100000,
                  },
                },
                spotlightClicks: false,
                disableBeacon: true,
                strict: false,
                floaterProps: {
                  disableAnimation: true,
                },
                content: {
                  title: "Quick Access",
                  desc: "Quick Access button",
                },
              },
            ]}
         

In the above JSON, I have passed actionCompleted key. Based on tat key value(true), I want to increase the stepindex and move to next slide and not before that.

 Is there a way to wait till the task is complete?. any help appreciated.

**Comment by gilbarbara** [comment]
hey @venumanthena 

It's too abstract to know for sure but have you tried to set a callback instead of a value, so you can execute it in the Joyride's callback.

```
{
  isReady: () => !isEmpty(get(reportData, "data", null)),
}
```

I can't be sure if this would work, but at least you'll have the current value.

**Reply by venumanthena** [reply]
not working sry.

**Reply by gilbarbara** [reply]
Anyway, the steps are created at the component initialization so I don't think that setting a value at that time will do what you need.
Try to execute the condition in the Joyride's callback.

**Reply by gilbarbara** [reply]
@venumanthena Did you fix it?

**Reply by venumanthena** [reply]
@gilbarbara Thanks for checking.  I haven't continued on it as I was on leave. Will let u know in few days time if it worked or not.
